### PR TITLE
[WIP]fixed definitions

### PR DESCRIPTION
**Pull request by kasecato**

### CODE REVIEW

**Comment by hedefalk** [comment]
I'm sorry, but the state of my branch here is basically a WIP for 
https://github.com/ensime/ensime-server/issues/1597

It's also based on master branch of ensime-node so the serverVersion attribute needs to be there.

File: src/features/definitions.ts [hunk 1]
```diff
@@ -1,26 +1,47 @@
...
-                   // TODO: How to we resolve a character offset into a row,col position before we have the buffer?
                    const uri = vscode.Uri.file(offsetPos.file)
-                   return new vscode.Location(uri, new vscode.Position(offsetPos.row-1, offsetPos.col))
+                   const openedDoc = vscode.workspace.textDocuments.find(doc => doc.fileName == offsetPos.file);
+                   if (openedDoc) {
+                       const pos = openedDoc.positionAt(offsetPos.offset)
+                       return new vscode.Location(uri, pos);
+                   } else {
+                       return vscode.workspace.openTextDocument(uri).then(definitionDoc => {
+                           vscode.window.showTextDocument(definitionDoc)
```
Comment (hedefalk) [inline src/features/definitions.ts]: I was thinking in these lines too, but isn't this going to open up a document everytime you just hover? You have the definitions hover thingy too so it happens way before an actual click…

Comment (kasecato) [reply src/features/definitions.ts]: I think so, too. This problem beats me.

**Comment by hedefalk** [comment]
Yeah, so I instead propose a fix in ensime-server to add line,col to
protocol. Info seems readily available...

**Pull request closed without merging**